### PR TITLE
Feature/workallocation/e UI 2986 task event completion

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2986
https://tools.hmcts.net/jira/browse/EUI-3548

### Change description ###
Support for tasks being completed when progressing cases through the various stages. At this point, there are three PACT stubs and a change to the case config file to set the URL for work allocation.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/services/ccd-config/ccd-case.config.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import {Injectable} from '@angular/core';
 import {AbstractAppConfig, CaseEditorConfig} from '@hmcts/ccd-case-ui-toolkit';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
 import {AppConfigService} from '../config/configuration.services';
+import { AppConstants } from '../../app.constants';
+import { AppUtils } from 'src/app/app-utils';
```
Comment (paul-graham) [inline src/app/services/ccd-config/ccd-case.config.ts:6]: This shouldn't be imported from `src/`.

File: src/app/services/ccd-config/ccd-case.config.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+import { WorkAllocationTaskService } from 'src/work-allocation/services';
```
Comment (paul-graham) [inline src/app/services/ccd-config/ccd-case.config.ts:7]: This shouldn't be imported from `src/`.

File: src/app/services/ccd-config/ccd-case.config.ts [hunk 3]
```diff
@@ -13,10 +17,21 @@ import {AppConfigService} from '../config/configuration.services';
 export class AppConfig extends AbstractAppConfig {
 
   protected config: CaseEditorConfig;
+  public workallocationUrl: string;
 
-  constructor(private appConfigService: AppConfigService) {
+  constructor(
+    private appConfigService: AppConfigService,
```
Comment (paul-graham) [inline src/app/services/ccd-config/ccd-case.config.ts:23]: Can you also make this one `readonly`? 😃

File: src/app/services/ccd-config/ccd-case.config.ts [hunk 4]
```diff
@@ -13,10 +17,21 @@ import {AppConfigService} from '../config/configuration.services';
...
+    ) {
     super();
     this.config = this.appConfigService.getEditorConfiguration() || {};
+    this.featureToggleWorkAllocation();
+  }
+
+  private featureToggleWorkAllocation(): void {
+    this.featureToggleService.isEnabled(AppConstants.FEATURE_NAMES.workAllocation).subscribe(
+      isFeatureEnabled => this.workallocationUrl = AppUtils.getFeatureToggledUrl(isFeatureEnabled, WorkAllocationTaskService.WorkAllocationUrl)
+      );
```
Comment (paul-graham) [inline src/app/services/ccd-config/ccd-case.config.ts:34]: Something odd going on with the indenting here...

File: api/workAllocation/routes.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Router } from 'express'
 
 import { getAllCaseWorkers, getAllCaseWorkersForLocation, getCaseWorkersForLocationAndService,
-        getCaseWorkersForService, getTask, postTaskAction, searchCaseWorker, searchTask } from '.'
+        getCaseWorkersForService, getTask, postTaskAction, postTaskSearchForCompletable, searchCaseWorker, searchTask } from '.'
```
Comment (paul-graham) [inline api/workAllocation/routes.ts:4]: This `import` set seems to stretch over two lines but not in a standard manner. If they don't fit on a single line, they should all be listed one per line.

**Review by paul-graham — CHANGES_REQUESTED** [review]
Some imports need some cleaning up and there's an indenting irregularity. Otherwise looks fine.

File: api/workAllocation/index.ts [hunk 6]
```diff
@@ -178,3 +179,22 @@ export async function searchCaseWorker(req: EnhancedRequest, res: Response, next
     next(error);
   }
 }
+
+export async function postTaskSearchForCompletable(req: EnhancedRequest, res: Response, next: NextFunction) {
+  try {
+    const postTaskPath: string = prepareTaskSearchForCompletable(baseWorkAllocationTaskUrl);
+    console.log('postTaskPath', postTaskPath)
```
Comment (connorpgpmcelroy) [inline api/workAllocation/index.ts:186]: Do we need to print this, and if so, why?

Comment (udaydenduluri33) [reply api/workAllocation/index.ts:186]: No removed it

File: api/workAllocation/routes.ts (hunk 5 above)
Comment (udaydenduluri33) [reply api/workAllocation/routes.ts:4]: Ok, moved it to different lines now.

File: src/app/services/ccd-config/ccd-case.config.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/app/services/ccd-config/ccd-case.config.ts]: yes, changed it

File: src/app/services/ccd-config/ccd-case.config.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/app/services/ccd-config/ccd-case.config.ts]: yes, changed it

File: src/app/services/ccd-config/ccd-case.config.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/app/services/ccd-config/ccd-case.config.ts]: yes done

File: src/app/services/ccd-config/ccd-case.config.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/app/services/ccd-config/ccd-case.config.ts:41]: Used Prettier now. fixed it

**Review by paul-graham — DISMISSED** [review]
Thank you for addressing all the points that were raised, @udaydenduluri33. Approved. 👍

**Review by connorpgpmcelroy — DISMISSED** [review]
LGTM

**Comment by udaydenduluri33** [comment]
This is already merged to master

**Pull request closed without merging**